prompt2model/dataset_generator/prompt_based.py

```diff
@@ -133,53 +139,33 @@ def construct_prompt(
         generation, it defaults to using the few_shot_example_string.
 
         The function uses different prompt templates based on the number of selected
-        examples from the generated dataset. If the total length of the prompt exceeds
-        3500 tokens, repeat the prompt generation process to generate a shorter one.
+        examples from the generated dataset.
 
         Args:
             instruction: The natural language instruction for the prompt.
             few_shot_example_string: A string representing the few-shot examples
                 parsed from the user's prompt, which quality is higher than the
-                genrated examples.
+                generated examples.
             generated_examples: A list of currently generated examples.
+            context_cutoff: If the total length of the prompt exceeds this value,
+                repeat the prompt generation process to generate a shorter one.
 
         Returns:
             The generated prompt string.
         """
-        # The random_example_string is a string, which contains several random
-        # few-shot examples as demonstrations for the DatasetGenerator. If
-        # generated_examples is empty, then the random_example_string
-        # is the few-shot examples parsed from the user's prompt.
         while True:
+            # Choose a few examples to add to the prompt if examples exist
```

Comment:
Until now we've been using the convention that comments should be complete, grammatical sentences.

```suggestion
            # Choose a few examples to add to the prompt if examples exist.
```